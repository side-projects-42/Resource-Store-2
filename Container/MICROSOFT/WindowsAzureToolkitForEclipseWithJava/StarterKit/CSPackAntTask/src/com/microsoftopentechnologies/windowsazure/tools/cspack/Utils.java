/**
 * Copyright (c) Microsoft Corporation
 * 
 * All rights reserved. 
 * 
 * MIT License
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files 
 * (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, 
 * publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, 
 * subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
 * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH 
 * THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */


package com.microsoftopentechnologies.windowsazure.tools.cspack;

import com.github.jknack.handlebars.*;
import com.github.jknack.handlebars.context.JavaBeanValueResolver;
import com.microsoftopentechnologies.windowsazure.tools.build.WindowsAzurePackage;
import org.apache.tools.ant.BuildException;
import org.apache.tools.ant.util.Base64Converter;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import javax.xml.bind.Unmarshaller;
import java.io.*;
import java.net.JarURLConnection;
import java.net.URL;
import java.net.URLConnection;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.*;
import java.util.jar.JarEntry;
import java.util.jar.JarFile;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

public class Utils {

    private static String convertByteArrayToHexString(byte[] arrayBytes) {
        StringBuffer stringBuffer = new StringBuffer();
        for (int i = 0; i < arrayBytes.length; i++) {
            stringBuffer.append(Integer.toString((arrayBytes[i] & 0xff) + 0x100, 16)
                    .substring(1));
        }
        return stringBuffer.toString();
    }

    /**
     * Calculate Sha256 hash
     */
    public static String calcHash(File file) throws IOException {
        int bufferSize = 16384;
        String output;
        try {
            BufferedInputStream is = new BufferedInputStream(new FileInputStream(file));
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte [] buffer = new byte[bufferSize];
            int sizeRead = -1;
            while ((sizeRead = is.read(buffer)) != -1) {
                digest.update(buffer, 0, sizeRead);
            }
            is.close();

            byte [] hash = null;
            hash = new byte[digest.getDigestLength()];
            hash = digest.digest();
            output = new Base64Converter().encode(hash);
        } catch (FileNotFoundException e) {
            throw new BuildException(e);
        } catch (NoSuchAlgorithmException e) {
            throw new BuildException(e);
        }
        return output;
    }

    public static String calcHexHash(File file) throws IOException {
        int bufferSize = 16384;
        String output;
        try {
            BufferedInputStream is = new BufferedInputStream(new FileInputStream(file));
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte [] buffer = new byte[bufferSize];
            int sizeRead = -1;
            while ((sizeRead = is.read(buffer)) != -1) {
                digest.update(buffer, 0, sizeRead);
            }
            is.close();

            byte [] hash = null;
            hash = new byte[digest.getDigestLength()];
            hash = digest.digest();
            output = convertByteArrayToHexString(hash).toUpperCase();
        } catch (FileNotFoundException e) {
            throw new BuildException(e);
        } catch (NoSuchAlgorithmException e) {
            throw new BuildException(e);
        }
        return output;
    }

    /**
     * Zip directory (source can't be a file)
     */
    public static void zipDirectory(File sourceDir, File zipFile, WindowsAzurePackage waPackage) {
        for (File file : sourceDir.listFiles()) {
            waPackage.zipFile(file, zipFile);
            delete(file);
        }
    }

    /**
     * Compresses the given directory and all its sub-directories into a ZIP file.
     * <p>
     * The ZIP file must not be a directory and its parent directory must exist.
     *
     * @param sourceDir
     *          root directory.
     * @param targetZipName
     *          name of ZIP file that will be created or overwritten.
     */
    public static void pack(File sourceDir, String targetZipName, boolean deleteSource) {
        if (!sourceDir.exists()) {
            throw new RuntimeException("Given file '" + sourceDir + "' doesn't exist!");
        }
        ZipOutputStream out = null;
        try {
            String[] filenames = sourceDir.list();
            if (filenames == null) {
                if (!sourceDir.exists()) {
                    throw new RuntimeException("Given file '" + sourceDir + "' doesn't exist!");
                }
                throw new IOException("Given file is not a directory '" + sourceDir + "'");
            }
            out = new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(new File(targetZipName))));
            pack(filenames, sourceDir, out, "");
            if (deleteSource) {
                for (String filename : filenames) {
                    delete(new File(sourceDir, filename));
                }
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        } finally {
            closeQuietly(out);
        }
    }

    /**
     * Compresses the given directory and all its sub-directories into a ZIP file.
     *
     * @param filenames
     *          files and directories names.
     * @param dir
     *          root directory.
     * @param out
     *          ZIP output stream.
     * @param pathPrefix
     *          prefix to be used for the entries.
     */
    private static void pack(String[] filenames, File dir, ZipOutputStream out, String pathPrefix) throws IOException {
        for (int i = 0; i < filenames.length; i++) {
            String filename = filenames[i];
            File file = new File(dir, filename);
            boolean isDir = file.isDirectory();
            String path = pathPrefix + file.getName();
            if (isDir) {
                path += "/";
            }
            // Create a ZIP entry
            ZipEntry zipEntry = new ZipEntry(path);
            if (!isDir) {
                zipEntry.setSize(file.length());
                zipEntry.setTime(file.lastModified());
            }
            out.putNextEntry(zipEntry);
            // Copy the file content
            if (!isDir) {
                copy(file, out);
            }
            out.closeEntry();
            // Traverse the directory
            if (isDir) {
                String[] dirFilenames = file.list();
                if (dirFilenames == null) {
                    if (!dir.exists()) {
                        throw new RuntimeException("Given file '" + dir + "' doesn't exist!");
                    }
                    throw new IOException("Given file is not a directory '" + dir + "'");
                }
                pack(dirFilenames, file, out, path);
            }
        }
    }

    /**
     * Copies the given file into an output stream.
     *
     * @param file input file (must exist).
     * @param out output stream.
     */
    public static void copy(File file, OutputStream out) throws IOException {
        FileInputStream in = new FileInputStream(file);
        try {
            byte[] buffer = new byte[4*1024];
            long count = 0;
            int n = 0;
            InputStream input = new BufferedInputStream(in);
            while (-1 != (n = input.read(buffer))) {
                out.write(buffer, 0, n);
                count += n;
            }
        }
        finally {
            closeQuietly(in);
        }
    }

    /**
     * Unconditionally close an <code>InputStream</code>.
     * @param stream the InputStream to close, may be null or already closed
     */
    public static void closeQuietly(Closeable stream) {
        try {
            if (stream != null) {
                stream.close();
            }
        } catch (IOException ioe) {
            // ignore
        }
    }

    public static void delete(File f) {
        if (f.isDirectory()) {
            for (File c : f.listFiles())
                delete(c);
        }
        if (!f.delete())
            throw new RuntimeException("Failed to delete file: " + f);
    }

    public static void generateFileList(Collection<File> fileList, File node) {
        //add file only
        if (node.isFile()) {
            fileList.add(node.getAbsoluteFile());
        }

        if (node.isDirectory()) {
            String[] subNote = node.list();
            for (String filename : subNote) {
                generateFileList(fileList, new File(node, filename));
            }
        }
    }

    public static String generateUID(String name) {
        UUID uuid = UUID.nameUUIDFromBytes(name.getBytes());
        return uuid.toString().replaceAll("-", "");
    }

    static void createDirectory(String name) {
        File serviceDefinitionDirectory = new File(name);
        if (!serviceDefinitionDirectory.exists()) {
            serviceDefinitionDirectory.mkdir();
        }
    }

    public static void applyTemplateWithPath(String templateName, Object model, String deployPath) throws IOException {
        applyTemplate(templateName, model, deployPath + File.separator + templateName);
    }

    public static void applyTemplate(String templateName, Object model, String fileName) throws IOException {
//        System.out.println("templateName=" + templateName + "; fileName=" + fileName);
        BufferedWriter fileWriter = new BufferedWriter(new FileWriter(fileName));
        try {
            Handlebars handlebars = new Handlebars();
            Template template = handlebars.compile("/templates/" + templateName);

            Context context = Context.newBuilder(model).resolver(JavaBeanValueResolver.INSTANCE).build();
            template.apply(context, fileWriter);
//            System.out.println(template.apply(context));
        } finally {
            fileWriter.flush();
            fileWriter.close();
        }
    }

    public static List<String> getJarEntries(String jarName, String entryName) throws IOException {
        List<String> files = new ArrayList<String>();
        System.out.println("entryName = " + entryName);
        JarURLConnection urlConnection = (JarURLConnection) new URL("jar:file:" + jarName + "!/" + entryName).openConnection();
//        URLConnection urlConnection = originUrl.openConnection();
            JarURLConnection jarConnection = ((JarURLConnection)urlConnection);
            JarFile jarFile = jarConnection.getJarFile();
            Enumeration<JarEntry> entries = jarFile.entries();

            while (entries.hasMoreElements()) {
                JarEntry entry = entries.nextElement();
                if (entry.getName().startsWith(jarConnection.getEntryName())) {
                    if (!entry.isDirectory()) {
                        files.add(entry.getName());
                    }
                }
            }
        return files;
    }

    public static void copyJarEntry(String source, File destination) throws IOException {
        InputStream stream = Utils.class.getResourceAsStream(source);
        if (stream == null) {
            throw new BuildException(source + " not found");
        }
        OutputStream resStreamOut = null;
        int readBytes;
        byte[] buffer = new byte[4096];
        try {
            resStreamOut = new FileOutputStream(destination);
            while ((readBytes = stream.read(buffer)) > 0) {
                resStreamOut.write(buffer, 0, readBytes);
            }
        } finally {
            stream.close();
            if (resStreamOut != null) {
                resStreamOut.close();
            }
        }
    }

    public static <T> T parseXmlFile(Class<T> type, String filePath) throws JAXBException {
        T instance = null;
        File file = new File(filePath);
        JAXBContext jaxbContext = JAXBContext.newInstance(type);

        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
        instance = (T) jaxbUnmarshaller.unmarshal(file);
        return instance;
    }
}
