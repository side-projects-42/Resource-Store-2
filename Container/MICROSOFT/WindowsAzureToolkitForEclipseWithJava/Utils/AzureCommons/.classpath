<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
	<classpathentry kind="lib" path="resources/ProjectManagerLibrary.jar" sourcepath="/ProjectManagerLibrary/src"/>
	<classpathentry combineaccessrules="false" kind="src" path="/com.microsoftopentechnologies.windowsazure.tools.sdk"/>
	<classpathentry kind="lib" path="resources/.cspack.jar"/>
	<classpathentry kind="output" path="bin"/>
</classpath>
