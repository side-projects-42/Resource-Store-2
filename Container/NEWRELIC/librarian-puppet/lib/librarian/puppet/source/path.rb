require 'librarian/source/path'
require 'librarian/puppet/source/local'

module Librarian
  module Puppet
    module Source
      class Path < Librarian::Source::Path
        include Local
      end
    end
  end
end
