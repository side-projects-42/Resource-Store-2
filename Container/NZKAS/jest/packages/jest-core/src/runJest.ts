/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import chalk from 'chalk';
import {sync as realpath} from 'realpath-native';
import {CustomConsole} from '@jest/console';
import {formatTestResults} from 'jest-util';
import exit from 'exit';
import fs from 'graceful-fs';
import {JestHook, JestHookEmitter} from 'jest-watcher';
import {Context} from 'jest-runtime';
import {Test} from 'jest-runner';
import {Config} from '@jest/types';
import {
  AggregatedResult,
  makeEmptyAggregatedTestResult,
} from '@jest/test-result';
import {ChangedFiles, ChangedFilesPromise} from 'jest-changed-files';
import getNoTestsFoundMessage from './getNoTestsFoundMessage';
import runGlobalHook from './runGlobalHook';
import SearchSource from './SearchSource';
import TestScheduler, {TestSchedulerContext} from './TestScheduler';
import TestSequencer from './TestSequencer';
import FailedTestsCache from './FailedTestsCache';
import collectNodeHandles from './collectHandles';
import TestWatcher from './TestWatcher';
import {TestRunData, Filter} from './types';

const getTestPaths = async (
  globalConfig: Config.GlobalConfig,
  context: Context,
  outputStream: NodeJS.WritableStream,
  changedFiles: ChangedFiles | undefined,
  jestHooks: JestHookEmitter,
  filter?: Filter,
) => {
  const source = new SearchSource(context);
  const data = await source.getTestPaths(globalConfig, changedFiles, filter);

  if (!data.tests.length && globalConfig.onlyChanged && data.noSCM) {
    new CustomConsole(outputStream, outputStream).log(
      'Jest can only find uncommitted changed files in a git or hg ' +
        'repository. If you make your project a git or hg ' +
        'repository (`git init` or `hg init`), Jest will be able ' +
        'to only run tests related to files changed since the last ' +
        'commit.',
    );
  }

  const shouldTestArray = await Promise.all(
    data.tests.map(test =>
      jestHooks.shouldRunTestSuite({
        config: test.context.config,
        duration: test.duration,
        testPath: test.path,
      }),
    ),
  );

  const filteredTests = data.tests.filter((_test, i) => shouldTestArray[i]);

  return {...data, allTests: filteredTests.length, tests: filteredTests};
};

type ProcessResultOptions = Pick<
  Config.GlobalConfig,
  'json' | 'outputFile' | 'testResultsProcessor'
> & {
  collectHandles?: () => Array<Error>;
  onComplete?: (result: AggregatedResult) => void;
  outputStream: NodeJS.WritableStream;
};

const processResults = (
  runResults: AggregatedResult,
  options: ProcessResultOptions,
) => {
  const {
    outputFile,
    json: isJSON,
    onComplete,
    outputStream,
    testResultsProcessor,
    collectHandles,
  } = options;

  if (collectHandles) {
    runResults.openHandles = collectHandles();
  } else {
    runResults.openHandles = [];
  }

  if (testResultsProcessor) {
    runResults = require(testResultsProcessor)(runResults);
  }
  if (isJSON) {
    if (outputFile) {
      const cwd = realpath(process.cwd());
      const filePath = path.resolve(cwd, outputFile);

      fs.writeFileSync(filePath, JSON.stringify(formatTestResults(runResults)));
      outputStream.write(
        `Test results written to: ${path.relative(cwd, filePath)}\n`,
      );
    } else {
      process.stdout.write(JSON.stringify(formatTestResults(runResults)));
    }
  }

  return onComplete && onComplete(runResults);
};

const testSchedulerContext: TestSchedulerContext = {
  firstRun: true,
  previousSuccess: true,
};

export default (async function runJest({
  contexts,
  globalConfig,
  outputStream,
  testWatcher,
  jestHooks = new JestHook().getEmitter(),
  startRun,
  changedFilesPromise,
  onComplete,
  failedTestsCache,
  filter,
}: {
  globalConfig: Config.GlobalConfig;
  contexts: Array<Context>;
  outputStream: NodeJS.WritableStream;
  testWatcher: TestWatcher;
  jestHooks?: JestHookEmitter;
  startRun: (globalConfig: Config.GlobalConfig) => void;
  changedFilesPromise?: ChangedFilesPromise;
  onComplete: (testResults: AggregatedResult) => void;
  failedTestsCache?: FailedTestsCache;
  filter?: Filter;
}) {
  const sequencer = new TestSequencer();
  let allTests: Array<Test> = [];

  if (changedFilesPromise && globalConfig.watch) {
    const {repos} = await changedFilesPromise;

    const noSCM = (Object.keys(repos) as Array<
      keyof ChangedFiles['repos']
    >).every(scm => repos[scm].size === 0);
    if (noSCM) {
      process.stderr.write(
        '\n' +
          chalk.bold('--watch') +
          ' is not supported without git/hg, please use --watchAll ' +
          '\n',
      );
      exit(1);
    }
  }

  const testRunData: TestRunData = await Promise.all(
    contexts.map(async context => {
      const matches = await getTestPaths(
        globalConfig,
        context,
        outputStream,
        changedFilesPromise && (await changedFilesPromise),
        jestHooks,
        filter,
      );
      allTests = allTests.concat(matches.tests);

      return {context, matches};
    }),
  );

  allTests = sequencer.sort(allTests);

  if (globalConfig.listTests) {
    const testsPaths = Array.from(new Set(allTests.map(test => test.path)));
    if (globalConfig.json) {
      console.log(JSON.stringify(testsPaths));
    } else {
      console.log(testsPaths.join('\n'));
    }

    onComplete && onComplete(makeEmptyAggregatedTestResult());
    return null;
  }

  if (globalConfig.onlyFailures && failedTestsCache) {
    allTests = failedTestsCache.filterTests(allTests);
    globalConfig = failedTestsCache.updateConfig(globalConfig);
  }

  const hasTests = allTests.length > 0;

  if (!hasTests) {
    const noTestsFoundMessage = getNoTestsFoundMessage(
      testRunData,
      globalConfig,
    );

    if (
      globalConfig.passWithNoTests ||
      globalConfig.findRelatedTests ||
      globalConfig.lastCommit ||
      globalConfig.onlyChanged
    ) {
      new CustomConsole(outputStream, outputStream).log(noTestsFoundMessage);
    } else {
      new CustomConsole(outputStream, outputStream).error(noTestsFoundMessage);

      exit(1);
    }
  } else if (
    allTests.length === 1 &&
    globalConfig.silent !== true &&
    globalConfig.verbose !== false
  ) {
    const newConfig: Config.GlobalConfig = {...globalConfig, verbose: true};
    globalConfig = Object.freeze(newConfig);
  }

  let collectHandles;

  if (globalConfig.detectOpenHandles) {
    collectHandles = collectNodeHandles();
  }

  if (hasTests) {
    await runGlobalHook({allTests, globalConfig, moduleName: 'globalSetup'});
  }

  if (changedFilesPromise) {
    testSchedulerContext.changedFiles = (await changedFilesPromise).changedFiles;
  }

  const results = await new TestScheduler(
    globalConfig,
    {startRun},
    testSchedulerContext,
  ).scheduleTests(allTests, testWatcher);

  sequencer.cacheResults(allTests, results);

  if (hasTests) {
    await runGlobalHook({allTests, globalConfig, moduleName: 'globalTeardown'});
  }

  return processResults(results, {
    collectHandles,
    json: globalConfig.json,
    onComplete,
    outputFile: globalConfig.outputFile,
    outputStream,
    testResultsProcessor: globalConfig.testResultsProcessor,
  });
});
