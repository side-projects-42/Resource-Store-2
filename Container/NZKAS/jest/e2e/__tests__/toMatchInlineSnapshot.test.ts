/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import fs from 'fs';
import path from 'path';
import {wrap} from 'jest-snapshot-serializer-raw';
import {cleanup, makeTemplate, writeFiles} from '../Utils';
import runJest from '../runJest';

const DIR = path.resolve(__dirname, '../to-match-inline-snapshot');
const TESTS_DIR = path.resolve(DIR, '__tests__');

const readFile = filename =>
  fs.readFileSync(path.join(TESTS_DIR, filename), 'utf8');

beforeEach(() => cleanup(TESTS_DIR));
afterAll(() => cleanup(TESTS_DIR));

test('basic support', () => {
  const filename = 'basic-support.test.js';
  const template = makeTemplate(
    `test('inline snapshots', () => expect($1).toMatchInlineSnapshot());\n`,
  );

  {
    writeFiles(TESTS_DIR, {
      [filename]: template(['{apple: "original value"}']),
    });
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('1 snapshot written from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('initial write');
  }

  {
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshots:   1 passed, 1 total');
    expect(stderr).not.toMatch('1 snapshot written from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot passed');
  }

  {
    writeFiles(TESTS_DIR, {
      [filename]: readFile(filename).replace('original value', 'updated value'),
    });
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshot name: `inline snapshots 1`');
    expect(status).toBe(1);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot mismatch');
  }

  {
    const {stderr, status} = runJest(DIR, [
      '-w=1',
      '--ci=false',
      filename,
      '-u',
    ]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('1 snapshot updated from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot updated');
  }
});

test('handles property matchers', () => {
  const filename = 'handle-property-matchers.test.js';
  const template = makeTemplate(`test('handles property matchers', () => {
      expect({createdAt: $1}).toMatchInlineSnapshot({createdAt: expect.any(Date)});
    });
    `);

  {
    writeFiles(TESTS_DIR, {[filename]: template(['new Date()'])});
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('1 snapshot written from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('initial write');
  }

  {
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshots:   1 passed, 1 total');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot passed');
  }

  {
    writeFiles(TESTS_DIR, {
      [filename]: readFile(filename).replace('new Date()', '"string"'),
    });
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshot name: `handles property matchers 1`');
    expect(stderr).toMatch('Snapshots:   1 failed, 1 total');
    expect(status).toBe(1);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot failed');
  }

  {
    writeFiles(TESTS_DIR, {
      [filename]: readFile(filename).replace('any(Date)', 'any(String)'),
    });
    const {stderr, status} = runJest(DIR, [
      '-w=1',
      '--ci=false',
      filename,
      '-u',
    ]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('1 snapshot updated from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('snapshot updated');
  }
});

test('removes obsolete external snapshots', () => {
  const filename = 'removes-obsolete-external-snapshots.test.js';
  const snapshotPath = path.join(
    TESTS_DIR,
    '__snapshots__',
    filename + '.snap',
  );
  const template = makeTemplate(`
    test('removes obsolete external snapshots', () => {
      expect('1').$1();
    });
  `);

  {
    writeFiles(TESTS_DIR, {[filename]: template(['toMatchSnapshot'])});
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('1 snapshot written from 1 test suite.');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('initial write');
    expect(fs.existsSync(snapshotPath)).toEqual(true);
  }

  {
    writeFiles(TESTS_DIR, {[filename]: template(['toMatchInlineSnapshot'])});
    const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshots:   1 obsolete, 1 written, 1 total');
    expect(status).toBe(1);
    expect(wrap(fileAfter)).toMatchSnapshot('inline snapshot written');
    expect(fs.existsSync(snapshotPath)).toEqual(true);
  }

  {
    const {stderr, status} = runJest(DIR, [
      '-w=1',
      '--ci=false',
      filename,
      '-u',
    ]);
    const fileAfter = readFile(filename);
    expect(stderr).toMatch('Snapshots:   1 file removed, 1 passed, 1 total');
    expect(status).toBe(0);
    expect(wrap(fileAfter)).toMatchSnapshot('external snapshot cleaned');
    expect(fs.existsSync(snapshotPath)).toEqual(false);
  }
});

test('supports async matchers', () => {
  const filename = 'async-matchers.test.js';
  const test = `
    test('inline snapshots', async () => {
      expect(Promise.resolve('success')).resolves.toMatchInlineSnapshot();
      expect(Promise.reject('fail')).rejects.toMatchInlineSnapshot();
    });
  `;

  writeFiles(TESTS_DIR, {[filename]: test});
  const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
  const fileAfter = readFile(filename);
  expect(stderr).toMatch('2 snapshots written from 1 test suite.');
  expect(status).toBe(0);
  expect(wrap(fileAfter)).toMatchSnapshot();
});

test('supports async tests', () => {
  const filename = 'async.test.js';
  const test = `
    test('inline snapshots', async () => {
      await 'next tick';
      expect(42).toMatchInlineSnapshot();
    });
  `;

  writeFiles(TESTS_DIR, {[filename]: test});
  const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
  const fileAfter = readFile(filename);
  expect(stderr).toMatch('1 snapshot written from 1 test suite.');
  expect(status).toBe(0);
  expect(wrap(fileAfter)).toMatchSnapshot();
});

test('writes snapshots with non-literals in expect(...)', () => {
  const filename = 'async.test.js';
  const test = `
    it('works with inline snapshots', () => {
      expect({a: 1}).toMatchInlineSnapshot();
    });
  `;

  writeFiles(TESTS_DIR, {[filename]: test});
  const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);

  const fileAfter = readFile(filename);
  expect(stderr).toMatch('1 snapshot written from 1 test suite.');
  expect(status).toBe(0);
  expect(wrap(fileAfter)).toMatchSnapshot();
});

// issue: https://github.com/facebook/jest/issues/6702
test('handles mocking native modules prettier relies on', () => {
  const filename = 'mockFail.test.js';
  const test = `
    jest.mock('path', () => ({}));
    jest.mock('fs', () => ({}));
    test('inline snapshots', () => {
      expect({}).toMatchInlineSnapshot();
    });
  `;

  writeFiles(TESTS_DIR, {[filename]: test});
  const {stderr, status} = runJest(DIR, ['-w=1', '--ci=false', filename]);
  expect(stderr).toMatch('1 snapshot written from 1 test suite.');
  expect(status).toBe(0);
});
