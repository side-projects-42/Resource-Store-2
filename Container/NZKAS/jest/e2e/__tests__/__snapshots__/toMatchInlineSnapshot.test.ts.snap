// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic support: initial write 1`] = `
test('inline snapshots', () =>
  expect({apple: 'original value'}).toMatchInlineSnapshot(\`
Object {
  "apple": "original value",
}
\`));

`;

exports[`basic support: snapshot mismatch 1`] = `
test('inline snapshots', () =>
  expect({apple: 'updated value'}).toMatchInlineSnapshot(\`
Object {
  "apple": "original value",
}
\`));

`;

exports[`basic support: snapshot passed 1`] = `
test('inline snapshots', () =>
  expect({apple: 'original value'}).toMatchInlineSnapshot(\`
Object {
  "apple": "original value",
}
\`));

`;

exports[`basic support: snapshot updated 1`] = `
test('inline snapshots', () =>
  expect({apple: 'updated value'}).toMatchInlineSnapshot(\`
Object {
  "apple": "updated value",
}
\`));

`;

exports[`handles property matchers: initial write 1`] = `
test('handles property matchers', () => {
  expect({createdAt: new Date()}).toMatchInlineSnapshot(
    {createdAt: expect.any(Date)},
    \`
Object {
  "createdAt": Any<Date>,
}
\`
  );
});

`;

exports[`handles property matchers: snapshot failed 1`] = `
test('handles property matchers', () => {
  expect({createdAt: "string"}).toMatchInlineSnapshot(
    {createdAt: expect.any(Date)},
    \`
Object {
  "createdAt": Any<Date>,
}
\`
  );
});

`;

exports[`handles property matchers: snapshot passed 1`] = `
test('handles property matchers', () => {
  expect({createdAt: new Date()}).toMatchInlineSnapshot(
    {createdAt: expect.any(Date)},
    \`
Object {
  "createdAt": Any<Date>,
}
\`
  );
});

`;

exports[`handles property matchers: snapshot updated 1`] = `
test('handles property matchers', () => {
  expect({createdAt: 'string'}).toMatchInlineSnapshot(
    {createdAt: expect.any(String)},
    \`
Object {
  "createdAt": Any<String>,
}
\`
  );
});

`;

exports[`removes obsolete external snapshots: external snapshot cleaned 1`] = `
test('removes obsolete external snapshots', () => {
  expect('1').toMatchInlineSnapshot(\`"1"\`);
});

`;

exports[`removes obsolete external snapshots: initial write 1`] = `

    test('removes obsolete external snapshots', () => {
      expect('1').toMatchSnapshot();
    });
  
`;

exports[`removes obsolete external snapshots: inline snapshot written 1`] = `
test('removes obsolete external snapshots', () => {
  expect('1').toMatchInlineSnapshot(\`"1"\`);
});

`;

exports[`supports async matchers 1`] = `
test('inline snapshots', async () => {
  expect(Promise.resolve('success')).resolves.toMatchInlineSnapshot(
    \`"success"\`
  );
  expect(Promise.reject('fail')).rejects.toMatchInlineSnapshot(\`"fail"\`);
});

`;

exports[`supports async tests 1`] = `
test('inline snapshots', async () => {
  await 'next tick';
  expect(42).toMatchInlineSnapshot(\`42\`);
});

`;

exports[`writes snapshots with non-literals in expect(...) 1`] = `
it('works with inline snapshots', () => {
  expect({a: 1}).toMatchInlineSnapshot(\`
Object {
  "a": 1,
}
\`);
});

`;
